### PR TITLE
Update frontend docs with TanStack Query example

### PR DESCRIPTION
**Pull request by simon-johansson**

### CODE REVIEW

File: docs/how-to/integrate-frontend.mdx [hunk 1]
```diff
@@ -49,6 +49,85 @@ console.log(response.text);
 
 See more in the [client generation docs](/docs/develop/client-generation).
 
+### Asynchronous state management
+
+When building something a bit more complex, you will likely need to deal with caching, refetching and data going stale.
```
Comment (marcuskohlberg) [inline docs/how-to/integrate-frontend.mdx:54]:
```suggestion
When building something a bit more complex, you will likely need to deal with caching, refetching, and data going stale.
```

File: docs/how-to/integrate-frontend.mdx [hunk 2]
```diff
@@ -49,6 +49,85 @@ console.log(response.text);
...
+[TanStack Query](https://tanstack.com/query/latest) is a popular library that was built to solve exactly these problems and works
+really well together the Encore request client.
```
Comment (marcuskohlberg) [inline docs/how-to/integrate-frontend.mdx:56]:
```suggestion
[TanStack Query](https://tanstack.com/query/latest) is a popular library that was built to solve exactly these problems and works well with the Encore request client.
```

File: docs/how-to/integrate-frontend.mdx [hunk 3]
```diff
@@ -49,6 +49,85 @@ console.log(response.text);
...
+}
+
+render(<App />, document.getElementById('root'))
+```
+
+This example assumes that we have a `todo` service with a `List` and `Add` endpoint. When adding the new todo,
+TanStack Query will automatically invalidate the `todos` query and refetch it.
+
+For a real-world example, take a look at the [Uptime Monitoring](https://github.com/encoredev/examples/tree/main/uptime) app which also makes use of
+TanStack Query´s `refetchInterval` option for polling the backend.
```
Comment (marcuskohlberg) [inline docs/how-to/integrate-frontend.mdx:129]:
```suggestion
TanStack Query's `refetchInterval` option for polling the backend.
```

File: docs/how-to/integrate-frontend.mdx [hunk 4]
```diff
@@ -49,6 +49,85 @@ console.log(response.text);
 
 See more in the [client generation docs](/docs/develop/client-generation).
 
+### Asynchronous state management
+
+When building something a bit more complex, you will likely need to deal with caching, refetching and data going stale.
+[TanStack Query](https://tanstack.com/query/latest) is a popular library that was built to solve exactly these problems and works
+really well together the Encore request client.
+
+Here is a simple example of using the Encore request client together with TanStack Query:
```
Comment (marcuskohlberg) [inline docs/how-to/integrate-frontend.mdx:58]:
```suggestion
Here is a simple example of using an Encore request client together with TanStack Query:
```

**Review by marcuskohlberg — APPROVED** [review]

File: docs/how-to/integrate-frontend.mdx [hunk 5]
```diff
@@ -49,6 +49,85 @@ console.log(response.text);
...
+    </QueryClientProvider>
+  )
+}
+
+function Todos() {
+  // Access the client
+  const queryClient = useQueryClient()
+
+  // Queries
+  const query = useQuery('todos', encoreClient.todo.List)
```
Comment (melkstam) [inline docs/how-to/integrate-frontend.mdx:89]: Just a heads up, the syntax in TanStack Query v5 (released just the other day) is a bit different 👀

https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#supports-a-single-signature-one-object

Comment (simon-johansson) [reply docs/how-to/integrate-frontend.mdx:89]: Great, thanks! Will update. Latest and greatest, here we go!

**Pull request merged**